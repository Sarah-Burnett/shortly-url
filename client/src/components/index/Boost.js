import React from "react";
import styled from "styled-components";
import { PrimaryButton, white, violet, alignPadding } from "../Styles";
import imgDesktop from "../../images/bg-boost-desktop.svg";
import imgMobile from "../../images/bg-boost-mobile.svg";

const BoostBox = styled.div`
	background: ${violet} url(${imgMobile}) no-repeat;
	width: 100vw;
	height: 300px;
	padding: 2vh ${alignPadding};
	margin: auto;
	color: ${white};
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	text-align: center;
	button {
		border-radius: 30px;
		padding: 2vh 5vw;
		margin-top: 4vh;
	}
	@media (min-width: 375px) {
		background-image: url(${imgDesktop});
		height: 250px;
	}
`;

const Boost = () => {
	return (
		<BoostBox>
			<h2>Boost your links today</h2>
			<PrimaryButton>
				<b>Get Started</b>
			</PrimaryButton>
		</BoostBox>
	);
};

export default Boost;
