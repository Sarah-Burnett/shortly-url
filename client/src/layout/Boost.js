import React from 'react'
import styled from 'styled-components'
import { PrimaryButton, white, violet } from '../Styles'

const BoostBox = styled.div`
    background: ${violet} url('../images/bg-boost-mobile.svg') no-repeat;
    width: 100vw;
    padding: 2vh 2vw;
    margin: auto;
    margin-bottom: 10vh;
    color: ${white};
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
    text-align: center;
    button {
        border-radius: 30px;
        padding: 2vh 5vw;
        margin: 2vh 0;
    }
`

export const Boost = () => {
    return (
        <BoostBox>
            <h1>Boost your links today</h1>
            <PrimaryButton><h2>Get Started</h2></PrimaryButton>
        </BoostBox>
    )
}
