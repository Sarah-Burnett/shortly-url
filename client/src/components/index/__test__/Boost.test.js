import React from 'react';
import ReactDOM from 'react-dom';
import Boost from '../Boost';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders boost without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Boost/>, div);
})

it('renders boost correctly', () => {
    const { getByRole } = render(<Boost/>);
    expect(getByRole('heading').textContent).toBe('Boost your links today')
    expect(getByRole('button').textContent).toBe('Get Started');
});

it('boost matches snapshot', () => {
    const tree = renderer.create(<Boost/>).toJSON();
    expect(tree).toMatchSnapshot();
})