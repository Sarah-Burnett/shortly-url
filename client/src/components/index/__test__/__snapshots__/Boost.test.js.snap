// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boost matches snapshot 1`] = `
<div
  className="sc-fzplWN jPNFiH"
>
  <h2>
    Boost your links today
  </h2>
  <button
    className="sc-AxirZ dVJbcR"
  >
    <b>
      Get Started
    </b>
  </button>
</div>
`;
